Core.ApplicationServices/GDPR/DataProcessingRegistrationApplicationService.cs

```diff
@@ -506,6 +506,32 @@ public Result<(DataProcessingRegistration registration, IEnumerable<DataProcessi
             });
         }
 
+        public Result<DataProcessingRegistration, OperationError> UpdateMainContract(int id, int contractId)
+        {
+            return Modify(id, registration =>
+            {
+                return registration.AssignMainContract(contractId)
+                    .Match
+                    (
+                        error => error,
+                        () => Result<DataProcessingRegistration, OperationError>.Success(registration)
+                    );
+            });
+        }
+
+        public Result<DataProcessingRegistration, OperationError> RemoveMainContract(int id, int contractId)
+        {
+            return Modify(id, registration =>
+            {
+                return registration.RemoveMainContract(contractId)
```

Comment:
there can only be one main contract.. why pass the contract id.. should it not just be a `ResetMainContract()`
. This one just opens up to error cases that we don't really need rights?